trino/client.py

```diff
@@ -334,11 +368,15 @@ def __call__(self, attempt):
 
 class _RetryWithExponentialBackoff(object):
     def __init__(
-            self, base=0.1, exponent=2, jitter=True, max_delay=2 * 3600  # 100ms  # 2 hours
-    ):
+        self,
+        base: float = 0.1,
+        exponent: int = 2,
+        jitter: bool = True,
+        max_delay: int = 2 * 3600,  # 100ms  # 2 hours
+    ) -> None:
         self._get_delay = _DelayExponential(base, exponent, jitter, max_delay)
 
-    def retry(self, func, args, kwargs, err, attempt):
+    def retry(self, func: Any, args: Any, kwargs: Any, err: Any, attempt: int) -> None:
```

Comment:
aren't kwargs always dicts?

Reply:
dictionary unpacking operator `**` is missing here, so `kwargs` parameter could be virtually anything. 
Parameters: `func`, `args`, `kwargs`, `err` are not used anywhere in this method anyway. Maybe I could just delete these not used parameters?

Reply:
I think the retry method was written very optimistically. I agree to simplify it to cover current usages only.

Reply:
done